CHANGELOG.md

```diff
@@ -4,6 +4,10 @@
 
 ### Snowpark Python API Updates
 
+#### Improvements
+
+- Reduced the number of additional [describe queries](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-example#retrieving-column-metadata) sent to the server to fetch the metadata of a DataFrame. It is still an experimental feature not enabled by default, and can be enabled by setting `session.reduce_describe_query_enabled` to `True`.
```

Comment:
Do we have any experimental results on the impact of this improvement?

Reply:
This is the first PR so I will try to run perf benchmark after completing https://snowflakecomputing.atlassian.net/browse/SNOW-1728988 and  https://snowflakecomputing.atlassian.net/browse/SNOW-1615885